File: backend/internal/vo/material.go
package vo

import "time"

type MaterialTreeNodeVO struct {
	ID         uint64               `json:"id"`
	ParentID   *uint64              `json:"parentId,omitempty"`
	Name       string               `json:"name"`
	Type       string               `json:"type"`
	FileExt    string               `json:"fileExt,omitempty"`
	MimeType   string               `json:"mimeType,omitempty"`
	FileSize   int64                `json:"fileSize,omitempty"`
	SortIndex  int                  `json:"sortIndex"`
	Children   []MaterialTreeNodeVO `json:"children,omitempty"`
	UpdatedAt  time.Time            `json:"updatedAt"`
}

type MaterialDetailVO struct {
	ID              uint64    `json:"id"`
	CourseID        uint64    `json:"courseId"`
	SpaceID         uint64    `json:"spaceId"`
	ParentID        *uint64   `json:"parentId,omitempty"`
	NodeType        string    `json:"nodeType"`
	NodeName        string    `json:"nodeName"`
	FileExt         string    `json:"fileExt,omitempty"`
	StoragePath     string    `json:"storagePath,omitempty"`
	MimeType        string    `json:"mimeType,omitempty"`
	FileSize        int64     `json:"fileSize"`
	LatestVersionNo int       `json:"latestVersionNo"`
	SortIndex       int       `json:"sortIndex"`
	CreatedBy       uint64    `json:"createdBy"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`
}

File: frontend/src/types/material.ts
export type MaterialNodeType = 'folder' | 'file'

export interface MaterialTreeNodeVO {
  id: number
  parentId?: number
  name: string
  type: MaterialNodeType
  fileExt?: string
  mimeType?: string
  fileSize?: number
  sortIndex: number
  updatedAt: string
  children?: MaterialTreeNodeVO[]
}

export interface MaterialDetailVO {
  id: number
  courseId: number
  spaceId: number
  parentId?: number
  nodeType: MaterialNodeType
  nodeName: string
  fileExt?: string
  storagePath?: string
  mimeType?: string
  fileSize: number
  latestVersionNo: number
  sortIndex: number
  createdBy: number
  createdAt: string
  updatedAt: string
}

export interface CreateMaterialFolderPayload {
  parentId?: number
  folderName: string
}

export interface UpdateMaterialNodePayload {
  nodeName: string
  sortIndex?: number
}
